queries.sql

```diff
@@ -14,4 +14,53 @@ SELECT * from animals WHERE neutered = TRUE;
 
 SELECT * from animals WHERE name != 'Gabumon';
 
-SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- Transactions: FOR ALL TRANSACTIONS I USED SELECT TO VERIFY THE STATUS AND TAKE SS.
+
+-- First
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Second
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals; 
+
+-- Third
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Fourth
+BEGIN TRANSACTION;
+DELETE from animals WHERE date_of_birth > '2022/01/01';
+SAVEPOINT animals_after_22;
+UPDATE animals SET weight_kg = (weight_kg * -1);
+SELECT * from animals;
+ROLLBACK TO SAVEPOINT animals_after_22;
+UPDATE animals SET weight_kg = (weight_kg * -1) WHERE weight_kg < 0;
+COMMIT;
+SELECT * from animals;
+
+-- Answer questions
+
+SELECT COUNT(*) from animals;
+
+SELECT COUNT(*) from animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) from animals;
+
+SELECT neutered, SUM(escape_attempts) from animals GROUP BY neutered;
```

Comment:
- [ ] You did a great job on most of the queries. :+1: Let's make some minor adjustments to get better results. Since we are trying to figure out which escapes the most, I believe `AVERAGE` is a better indicator because the number of neutered and non-neutered animals may vary and the sum might be influenced by the respective numbers. Kindly use `AVG` instead of `SUM`. ✍️